BizHawk.Client.EmuHawk/ArgParser.cs

```diff
@@ -135,39 +135,63 @@ public void ParseArguments(string[] args)
 				{
 					cmdRom = args[i];
 				}
-			}
-			////initialize HTTP communication
+			}
+
+			//initialize HTTP communication
 			if (URL_get != null || URL_post != null)
 			{
+				GlobalWin.httpCommunication = new Communication.HttpCommunication();
 				if (URL_get != null)
 				{
-					GlobalWin.httpCommunication.initialized = true;
 					GlobalWin.httpCommunication.SetGetUrl(URL_get);
 				}
 				if (URL_post != null)
 				{
-					GlobalWin.httpCommunication.initialized = true;
 					GlobalWin.httpCommunication.SetPostUrl(URL_post);
 				}
-			}
-			//inititalize socket server
-			if (socket_ip != null && socket_port > -1)
-			{
-				GlobalWin.socketServer.initialized = true;
+			}
+
+			//inititalize socket server
+			if (socket_ip != null && socket_port > 0)
+			{
+				GlobalWin.socketServer = new Communication.SocketServer();
 				GlobalWin.socketServer.SetIp(socket_ip, socket_port);
 			}
-			else if (socket_ip != null)
+			else if (socket_ip == null ^ socket_port == 0)
 			{
-				GlobalWin.socketServer.initialized = true;
-				GlobalWin.socketServer.SetIp(socket_ip);
-			}
-			
-			//initialize mapped memory files
+				throw new ArgParserException("Socket server needs both --socket_ip and --socket_port. Socket server was not started");
+			}
+
+			//initialize mapped memory files
 			if (mmf_filename != null)
 			{
-				GlobalWin.memoryMappedFiles.initialized = true;
+				GlobalWin.memoryMappedFiles = new Communication.MemoryMappedFiles();
 				GlobalWin.memoryMappedFiles.SetFilename(mmf_filename);
 			}
 		}
+
+		public static string GetCmdConfigFile(string[] args)
```

Comment:
method can be
```csharp
return args.FirstOrDefault(arg => arg.StartsWith("--config=", StringComparison.InvariantCultureIgnoreCase));
    ?.Substring(9);
```

Reply:
Thanks for the suggestion, will be changed accordingly.